runtime/j9vm/j9memcategories.c

```diff
@@ -39,11 +39,19 @@
 
 /* Note: OMRMEM_CATEGORY_UNKNOWN and OMRMEM_CATEGORY_PORT_LIBRARY are stored inside the port library */
 OMRMEM_CATEGORY_4_CHILDREN("JRE", J9MEM_CATEGORY_JRE, OMRMEM_CATEGORY_VM, OMRMEM_CATEGORY_JIT, J9MEM_CATEGORY_CLASS_LIBRARIES, OMRMEM_CATEGORY_UNKNOWN);
+#if JAVA_SPEC_VERSION >= 16
+#if defined(OMR_OPT_CUDA)
+OMRMEM_CATEGORY_10_CHILDREN("VM", OMRMEM_CATEGORY_VM, J9MEM_CATEGORY_CLASSES, J9MEM_CATEGORY_MODULES, OMRMEM_CATEGORY_MM, OMRMEM_CATEGORY_THREADS, OMRMEM_CATEGORY_TRACE, J9MEM_CATEGORY_JVMTI, J9MEM_CATEGORY_JNI, OMRMEM_CATEGORY_CUDA, OMRMEM_CATEGORY_PORT_LIBRARY, J9MEM_CATEGORY_VM_FFI);
+#else /* OMR_OPT_CUDA */
+OMRMEM_CATEGORY_9_CHILDREN("VM", OMRMEM_CATEGORY_VM, J9MEM_CATEGORY_CLASSES, J9MEM_CATEGORY_MODULES, OMRMEM_CATEGORY_MM, OMRMEM_CATEGORY_THREADS, OMRMEM_CATEGORY_TRACE, J9MEM_CATEGORY_JVMTI, J9MEM_CATEGORY_JNI, OMRMEM_CATEGORY_PORT_LIBRARY, J9MEM_CATEGORY_VM_FFI);
```

Comment:
J9MEM_CATEGORY_VM_FFI still also needs to be declared itself as having no children. As an example see J9MEM_CATEGORY_JNI.

Reply:
Once this is fixed, with a build of the latest code, after memory is allocated for J9MEM_CATEGORY_VM_FFI, dump a javacore file and look at the "NATIVEMEMINFO subcomponent" to see how J9MEM_CATEGORY_VM_FFI fits into the tree.

Reply:
`J9MEM_CATEGORY_VM_FFI` has been declared as having no children and added to the category table.